backend/pluto/pluto/_internal/adapters/storemgr.py

```diff
@@ -95,3 +78,41 @@ def _fmtsqllit(self, val: Any) -> str:
             return f"'{val}'"
         else:
             return val
+
+
+class PGSQLStorageManager(SQLStorageManager):
+    def __init__(self, cfg: Config) -> None:
+        super().__init__(cfg)
+        self._connstr = "user={} password={} host={} port={} dbname={}".format(
+            self._cfg.dbuser,
+            self._cfg.dbpassword,
+            self._cfg.dbhost,
+            self._cfg.dbport,
+            self._cfg.dbname,
+        )
+
+    # Methods that must be overrided
+    def query(self, q: str) -> list[Any]:
+        logger.debug("Querying database with string: {}".format(q))
+        if self._conn:
+            cur = self._conn.cursor()
+            cur.execute(q)
+            self._conn.commit()
+            rows_clean: list[tuple[str]] = []
+            if "SELECT" in q and cur.rowcount > 0:
+                rows = cur.fetchall()
+                rows_clean = list(map(lambda r: tuple(map(lambda s: s.strip(), r)), rows))  # type: ignore
+            cur.close()
+            return rows_clean
+        else:
+            raise ValueError(
+                "Database has no connection! Must explicit call connect() before queries!"
+            )
+
+    def connect(self):
+        logger.debug(f"Connecting to {self._cfg.dbname}")
+        self._conn = psycopg2.connect(self._connstr)
+
+    def close_conn(self):
```

Comment:
I know it was not there before, but maybe we can add a call to this function to the destructor:

```python
def __del__(self):
  if self._conn:
    self._conn.close()
```